psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1519,47 +1672,67 @@ void OEProp::compute_mo_extents()
             double xx = quadrupole[0]->get(0, i),
                    yy = quadrupole[1]->get(0, i),
                    zz = quadrupole[2]->get(0, i);
-            outfile->Printf( "    %4d%3s%3d%15.10f%15.10f%15.10f%15.10f\n",
-                    i,
-                    labels[h].c_str(),
-                    n,
-                    std::fabs(quadrupole[0]->get(0, i)),
-                    std::fabs(quadrupole[1]->get(0, i)),
-                    std::fabs(quadrupole[2]->get(0, i)),
-                    std::fabs(xx + yy + zz));
-                    mo_es[0]->set(0,i,quadrupole[0]->get(0, i));
-                    mo_es[1]->set(0,i,quadrupole[1]->get(0, i));
-                    mo_es[2]->set(0,i,quadrupole[2]->get(0, i));
-                    mo_es[3]->set(0,i,fabs(xx + yy + zz));
+            if (print_output) {
+                outfile->Printf("    %4d%3s%3d%15.10f%15.10f%15.10f%15.10f\n", i, labels[h].c_str(), n,
+                                std::fabs(quadrupole[0]->get(0, i)), std::fabs(quadrupole[1]->get(0, i)),
+                                std::fabs(quadrupole[2]->get(0, i)), std::fabs(xx + yy + zz));
+            }
+            mo_es[0]->set(0, i, quadrupole[0]->get(0, i));
+            mo_es[1]->set(0, i, quadrupole[1]->get(0, i));
+            mo_es[2]->set(0, i, quadrupole[2]->get(0, i));
+            mo_es[3]->set(0, i, fabs(xx + yy + zz));
+        }
+        if (print_output) {
+            outfile->Printf("\n");
         }
-
-        outfile->Printf( "\n");
 
     } else {
 
         // TODO: Both alpha and beta orbitals are reported separately
         // This helps identify symmetry breaking
     }
-    wfn_->set_mo_extents(mo_es);
+    return mo_es;
+}
+
+typedef PopulationAnalysisCalc PAC;
+PopulationAnalysisCalc::PopulationAnalysisCalc(std::shared_ptr<Wavefunction> wfn) : Prop(wfn) {
+    // No internal state. num_noon is now an argument.
 }
 
+PopulationAnalysisCalc::~PopulationAnalysisCalc() {}
+
 void OEProp::compute_mulliken_charges()
 {
-    outfile->Printf( "  Mulliken Charges: (a.u.)\n");
+    PAC::SharedStdVector Qa_ptr, Qb_ptr, apcs;
+    std::tie(Qa_ptr, Qb_ptr, apcs) = pac_.compute_mulliken_charges(true);
+    wfn_->set_atomic_point_charges(apcs);
+
+    auto vec_apcs = std::make_shared<Matrix>("Mulliken Charges: (a.u.)", 1, apcs->size());
+    for (size_t i = 0; i < apcs->size(); i++) {
+        vec_apcs->set(0, i, (*apcs)[i]);
+    }
+    wfn_->set_array("MULLIKEN_CHARGES", vec_apcs);
+}
 
+std::tuple<PAC::SharedStdVector, PAC::SharedStdVector, PAC::SharedStdVector>
+PopulationAnalysisCalc::compute_mulliken_charges(bool print_output) {
+    if (print_output) {
+        outfile->Printf("  Mulliken Charges: (a.u.)\n");
+    }
     std::shared_ptr<Molecule> mol = basisset_->molecule();
+
+    auto Qa_ptr = std::make_shared<std::vector<double>>(mol->natom());
+    auto Qb_ptr = std::make_shared<std::vector<double>>(mol->natom());
+    std::vector<double>& Qa = *Qa_ptr;
```

Comment:
On this we would typically name `Qa_ptr` as Qa and then `double* Qa_ptr = Qa.data()`.